tests/unit/test_target_filtering.py

```
@@ -0,0 +1,644 @@
+
+    def test_no_targets_extractor_forwarded_to_tasks(self):
+        """Scan-level targets_ (port type) must NOT appear in individual task opts."""
+        with patch('secator.runners.task.discover_tasks', side_effect=patched_discover_tasks):
+            _, chain_sig = self._build_workflow2_chain(chain_previous_results=True)
+            all_opts = get_all_task_opts(chain_sig)
+            # w2_2 has no task-level targets_ extractor, so it must not have targets_ at all
+            w2_2_opts_list = [o for o in all_opts if o.get('name') == 'w2_2']
+            self.assertGreaterEqual(len(w2_2_opts_list), 1, "w2_2 opts not found in chain")
+            for opts in w2_2_opts_list:
+                self.assertNotIn('targets_', opts,
+                    f"Task w2_2 must not receive scan-level targets_ extractor")
```

Flake8 (7.3.0)

[error] f-string is missing placeholders

(F541)